File: backend/notifications/src/main/java/com/oxygensend/notifications/config/MailProperties.java
package com.oxygensend.notifications.config;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;

@Validated
@ConfigurationProperties(prefix = "mail")
public record MailProperties(@NotEmpty
                             String host,
                             @NotNull
                             int port,
                             String username,
                             String password,
                             @NotEmpty
                             String protocol,
                             boolean auth,
                             boolean starttlsEnable,
                             boolean debug,
                             boolean testConnection,
                             @NotEmpty
                             String emailFrom
) {

}

File: backend/notifications/src/main/kotlin/com/oxygensend/notifications/config/MailProperties.kt
package com.oxygensend.notifications.config

import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.NotNull
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated

@Validated
@ConfigurationProperties(prefix = "mail")
data class MailProperties(
    val host: @NotEmpty String,
    val port: @NotNull Int,
    val username: String?,
    val password: String?,
    val protocol: @NotEmpty String,
    val auth: Boolean,
    val starttlsEnable: Boolean,
    val debug: Boolean,
    val testConnection: Boolean,
    val emailFrom: @NotEmpty String
)
